addons/web_editor/static/src/js/rte.js

```diff
@@ -533,9 +533,8 @@ var RTE = Widget.extend({
             }, 150); // setTimeout to remove flickering when change to editable zone (re-create an editor)
             this.$last = null;
         }
-        if ($editable.length && (!this.$last || this.$last[0] !== $editable[0]) &&
-                ($target.closest('[contenteditable]').attr('contenteditable') || "").toLowerCase() !== 'false') {
-
+        if (($editable.length && ($target.closest('[contenteditable]').attr('contenteditable') || "").toLowerCase() !== 'false' || dom.isImg($target[0]))
+            && (!this.$last || this.$last[0] !== $editable[0])) {
```

Comment:
Yes this is because all `contenteditable` attributes are removed when saving. If parts of a snippet cannot be edited you have to put the `o_not_editable` class on it (instead of `contenteditable="false"`). But I never tested it in details.
However in this case, that is JS code for the snippet option that controls the edition, so you have to use `contenteditable`. Are there problems with my suggested fix ? :)